### ISSUE TITLE
Signer is not sent an activity notification when signature is requested

### ISSUE THREAD

**Issue by DazeEnd** [LibreSign/libresign#2425]
**Describe the bug**
A user with a Nextcloud account should receive a notification when his signature is requested. Currently, no notification is sent.

**To Reproduce**
1. As a regular Nextcloud user, log into your account. Open Settings > Notifications, and make sure that the "Mail" checkbox is selected for all notification options, and that "As soon as possible" is selected for the "Send activity emails" option.
![image](https://github.com/LibreSign/libresign/assets/2699247/6708abc0-2b47-45ab-9f60-2af3537e1c7d)

2. Log in as an admin user and request a signature from the regular Nextcloud user from Step 1.

3. Wait 10 minutes.

**Expected behavior**
The regular user whose signature was requested should receive an email notification informing him that his signature has been requested.

**Actual behavior**
No email notification is sent. This was verified by looking at the logs of my SMTP server.

**Screenshots**
![image](https://github.com/LibreSign/libresign/assets/2699247/6459fafd-5d97-45a4-b43f-ee7fd2483a78)

**Environment information (please complete the following information):**
 - OS: AlmaLinux 8.9
 - Browser: Chrome
 - LibreSign Version: v8.0.0-rc6
 - Nextcloud Server Version: v28.0.2
 - Logs (get the entries from nextcloud.log related with LibreSign i.e `tail -f data/nextcloud.log|grep libresign`): No messages are logged when performing these steps

**Additional context**
When a signature is requested from an email address that does *not* have a Nextcloud account, that email address *does* receive an email notification.

**Comment by DazeEnd** [comment]
Tested in LibreSign 8.0.0-rc7 running on Nextcloud 28.0.2.

I logged in as the regular user, went to Settings > Notifications, and turned on the new "You have a file to sign" notfication:
![image](https://github.com/LibreSign/libresign/assets/2699247/b247095f-b5fd-4585-9816-1c51826a0451)

Logging into Nextcloud as the admin user, I requested a signature from the regular user and waited for a notification to arrive via email. After a few minutes, this email arrived:
![image](https://github.com/LibreSign/libresign/assets/2699247/59d440ce-2832-4bf5-92f4-d0e99d777453)

**Although the email did arrive, it did not contain the information that I expected.** As you can see, the email only says that "There was some activity at Nextcloud". At minimum, I expected the email to indicate that the activity that I am being notified about is that there is a new file to sign, and indicate what the name of the file is. It would also be great if there was a button that took the user to the file, but that is not strictly necessary.

In general, I would expect the notification email to either look like all the other activity notification emails that are sent by Nextcloud, or like the existing notification emails that LibreSign sends when a signature is requested from an email address 

Here are some mockups:

**Mock-up: LibreSign notification modeled on other activity notifications.** (This is my preferred solution, since it integrates with with activity notifications the user is already receiving.)
![image](https://github.com/LibreSign/libresign/assets/2699247/c1cb1f18-195f-41f2-a75e-1d5acc4e92fb)

**Mock-up: LibreSign notification modeled on existing LibreSign emails.**
![Screenshot from 2024-03-08 16-06-34](https://github.com/LibreSign/libresign/assets/2699247/d9fb0ee6-fe9f-4abc-9734-512b4b7ecd6c)

**Comment by DazeEnd** [comment]
Looking more closely at the emails, it appears that you might have already been attempting to use the standard activity emails, as in the first mock-up above. However, it looks like the activity didn't properly register.

In case it is important, I will point out that LibreSign did not add an activity notification in the Activities app either:
![image](https://github.com/LibreSign/libresign/assets/2699247/a34e9150-9953-40b6-99f5-27fa7642170c)

**Comment by vitormattos** [comment]
I reopened this issue to check more.

In my environment I'm receiving this email:

![Screenshot_20240308_182247](https://github.com/LibreSign/libresign/assets/1079143/4e08bc72-78e2-4217-b9e6-c94128d92af3)

**Comment by DazeEnd** [comment]
I just shared a file to see if I could get Files to generate an activity email, and everything seems to be working as expected outside of LibreSign:
![image](https://github.com/LibreSign/libresign/assets/2699247/f873d8c0-6cda-446d-af1c-ab34b3ff7257)

I just checked, and the string "libresign" does not appear in `nextcloud.log`.

**Comment by vitormattos** [comment]
Is working to me:

![Screenshot_20240308_182928](https://github.com/LibreSign/libresign/assets/1079143/dee0ab64-6781-40b0-8bbc-a593a2becdae)

**Comment by DazeEnd** [comment]
Let me turn off SELinux, reboot, and try again.

**Comment by vitormattos** [comment]
Did you enabled here?

![Screenshot_20240308_183154](https://github.com/LibreSign/libresign/assets/1079143/e82fb2dc-affa-414c-a6e5-99f4ab30df60)

**Comment by DazeEnd** [comment]
Yes, the "You have a file to sign" activity is turned ON:
![image](https://github.com/LibreSign/libresign/assets/2699247/875ede5c-e9fa-470e-a2dc-abc005c06b3b)

I have turned off SELinux and rebooted:
```
[ceperry@newton ~]$ sestatus
SELinux status:                 disabled
```

I uploaded a new document and requested a signature on the document. My next notification email showed the files activity of uploading the file, but there was no activity from LibreSign:
![image](https://github.com/LibreSign/libresign/assets/2699247/0e1cb595-3a2c-48ef-88cc-37a05d467e24)

**Comment by vitormattos** [comment]
Not related about email, but I identified that Activity app have a class called Filter to be possible display the activities at Activity app. I'm implementing this now, maybe will help to check if you are receiving the notifications.

**Comment by vitormattos** [comment]
![Screenshot_20240308_191042](https://github.com/LibreSign/libresign/assets/1079143/c284ddb9-52c5-4f30-924f-9d48f7f2701c)

**Comment by vitormattos** [comment]
@DazeEnd could you check if is ok looking the screenshots here #2509 ?
I will wait your review to don't send new words to translators because this PR changed and created texts.

**Comment by DazeEnd** [comment]
@vitormattos Sorry for the delay in getting back to you. The screenshots look good to me.

The notifications include the message, "admin requested your signature on document." In this message, will "document" be replaced with the name of the file? For example, "admin requested your signature on Perry Contract.pdf"? If yes, then it's fine.

However, if "document" will _not_ be replaced, and it will always read "[username] requested your signature on document", then I think it should be slightly reworded to say "[username] requested your signature on **a** document".

**Comment by vitormattos** [comment]
The document name in my case is `document` 😅
Low creativity for a decent name.

**Comment by DazeEnd** [comment]
Testing on RC8, this problem seems to be mostly fixed, but not completely fixed.

When I request a signature from a user with a Nextcloud account, the user whose signature was requested receives the following notifications:

1. The user immediately receives an email informing him that his signature has been requested. (See "Screenshot 1", below.)
2. The user later receives an activity email notifying the user that LibreSign made changes to a document. (See "Screenshot 2", below.)
3. The user has an activity posted in the Activity app, saying that the requesting user made changes to a document. (See "Screenshot 3", below.)

What still seems to be missing is an activity indicating that someone "requested your signature" on the document, as illustrated in screenshots in #2509. (See "Screenshots from Issue 2509", below.)

**Screenshot 1:**
![image](https://github.com/LibreSign/libresign/assets/2699247/9b3ee0f5-ac86-4905-9372-2820672df2d5)

**Screenshot 2:**
![image](https://github.com/LibreSign/libresign/assets/2699247/65f7c26a-6ae9-404d-b87d-9c73a4b312b8)

**Screenshot 3:**
![Screenshot from 2024-03-18 16-50-51](https://github.com/LibreSign/libresign/assets/2699247/bad85b90-476a-4744-9b5a-76527c06aa14)

**Screenshots from Issue 2509:**
![image](https://github.com/LibreSign/libresign/assets/2699247/a0bd875e-4caa-4d16-96e7-4be87a9ee5f4)
![image](https://github.com/LibreSign/libresign/assets/2699247/38cfc379-7519-45fb-a9ae-5c894b80b727)
![image](https://github.com/LibreSign/libresign/assets/2699247/1fd26bad-39e4-49fa-a3ec-5fc5f55a9fe1)

**Comment by vitormattos** [comment]
Hi, I will reopen this issue and do a best check.

**Comment by vitormattos** [comment]
Fixed by:
- #2573

**Comment by vitormattos** [comment]
Fixed at: https://github.com/LibreSign/libresign/releases/download/v8.0.0-rc9/libresign-v8.0.0-rc9.tar.gz

**Comment by DazeEnd** [comment]
This is fixed for me in v8.0.0-rc9.
